### PR TITLE
base code for level updater

### PR DESCRIPTION
**Pull request by bbeck13**

### CODE REVIEW

File: src/game_updater/LevelUpdater.cpp [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+  this->position = 0;
+}
 LevelUpdater::~LevelUpdater() {}
 
-void LevelUpdater::Update(std::shared_ptr<Level> game_state) {}
+void LevelUpdater::Update(std::shared_ptr<Level> level) {
+  if (position == 0) {
+    position = 1;
+    level->getMusic()->play();
+    level->getMusic()->setLoop(false);
```
Comment (josepharhar) [inline src/game_updater/LevelUpdater.cpp:12]: Is calling play() once every update a good idea?

File: src/RhythmRunner.cpp [hunk 2]
```diff
@@ -79,16 +82,27 @@ static void ResizeCallback(GLFWwindow* window, int width, int height) {
...
+  double clock = glfwGetTime();
   while (!glfwWindowShouldClose(renderer->GetWindow())) {
-    renderer->Render(NULL);
+    double nextTime = glfwGetTime();
+    if (nextTime - clock > SEC_PER_FRAME) {
+      updater->Update(gameState);
+      renderer->Render(gameState);
+
+      clock = nextTime;
+    }
```
Comment (josepharhar) [inline src/RhythmRunner.cpp:105]: I have other thoughts on how to do the time stepping, but this works for now

File: src/game_updater/LevelUpdater.cpp (hunk 1 above)
Comment (bbeck13) [reply src/game_updater/LevelUpdater.cpp:12]: thats just on the first call to update I added
`#define START 0
#define PLAYING 1
#define END 2
`
in a latter commit I'll push that diff in  sec

Comment (josepharhar) [reply src/game_updater/LevelUpdater.cpp:12]: :ok_hand:

**Comment by josepharhar** [comment]
CR 🔧 QA 🔨

**Pull request merged**